File: snail.java
public class snail 
{
    public static void main(String[] args) 
    {   
        //Flag for reaching top of pole.
        boolean reachedTopOfPole = false;
        //Current snail height in meters.
        int currentHeight = 0;
        //current day of climb.
        int currentDay = 1;
        //Height of pole.
        int h = 10;
        //Ascend during day.
        int a = 3;
        //Descend during night.
        int b = 2;

        //While we have not reached the top.
        while (reachedTopOfPole != true)
        {
            if (currentHeight + a >= h)
            {
                System.out.println("Ladies and gentlemen. The snail has reached the summit.");
                reachedTopOfPole = true;
            }
            else
            {
                currentHeight += (a - b);
                System.out.println("Day " + currentDay + " height is " + currentHeight + " meters");
                currentDay += 1;
            }
        }
        System.out.println("The snail has reached the top of the pole on day " + currentDay);
    }
}

File: snail.py
reachedTopOfPole = False
currentHeight = 0
currentDay = 1       
h = 10
a = 3
b = 2
#While we have not reached the top.
while reachedTopOfPole != True:
    if currentHeight + a >= h:
        print("Ladies and gentlemen. The snail has reached the summit.")
        reachedTopOfPole = True
    else:
        currentHeight += (a - b);
        currentDay += 1;
print("The snail has reached the top of the pole on day ",currentDay);